packages/pxweb2-ui/src/lib/components/SearchSelect/SearchSelect.tsx

```diff
@@ -0,0 +1,247 @@
+        );
+        break;
+      case 'Enter':
+        if (
+          isOpen &&
+          highlightedIndex >= 0 &&
+          highlightedIndex < filteredOptions.length
+        ) {
+          handleSelect(filteredOptions[highlightedIndex]);
+        } else if (exactMatch) {
+          handleSelect(exactMatch);
+        }
+        break;
+      case 'Tab':
+        confirmSelection();
+        break;
+      case 'Escape':
+        setIsOpen(false);
+        setHighlightedIndex(-1);
+        break;
+      default:
+        if (!isOpen && (e.key === 'ArrowDown' || e.key === 'ArrowUp')) {
+          setIsOpen(true);
+        }
+        break;
+    }
+  };
+
+  const confirmSelection = () => {
+    const trimmed = inputValue.trim().toLowerCase();
+    const match = options.find((opt) => opt.label.toLowerCase() === trimmed);
+
+    if (match) {
+      handleSelect(match);
+    } else {
+      setInputValue('');
+    }
+
+    setIsOpen(false);
+    setHighlightedIndex(-1);
+  };
+
+  const showClearButton = !!selectedOption || inputValue.length > 0;
+
+  return (
+    <div className={styles.searchableSelect}>
+      {label && (
+        <Label
+          className={styles.label}
+          id={`${searchSelectId}-label`}
+          htmlFor={`${searchSelectId}-combobox`}
+          size="medium"
+        >
+          {label}
+        </Label>
+      )}
+      <div className={styles.contentWrapper} ref={contentRef}>
+        <input
+          id={`${searchSelectId}-combobox`}
+          ref={inputRef}
+          type="text"
+          autoComplete="off"
+          className={cl(styles.input, styles['bodyshort-medium'])}
+          placeholder={placeholder}
+          value={inputValue}
+          onClick={() => setIsOpen(true)}
+          onChange={(e) => {
+            setInputValue(e.target.value);
+            setIsOpen(true);
+            setHighlightedIndex(-1);
+          }}
+          onFocus={() => setIsOpen(true)}
+          onKeyDown={handleKeyDown}
+          onBlur={() => setTimeout(confirmSelection, 100)}
+          role="combobox"
+          inputMode={inputMode}
+          pattern={inputMode === 'numeric' ? '[0-9]*' : undefined}
+          aria-autocomplete="list"
+          aria-haspopup="listbox"
+          aria-expanded={isOpen}
+          aria-controls={`${searchSelectId}-listbox`}
+          aria-activedescendant={
+            highlightedIndex >= 0
+              ? `${searchSelectId}-option-${highlightedIndex}`
+              : undefined
+          }
+        />
+        <div className={styles.iconWrapper}>
+          {showClearButton ? (
+            <Button
+              variant="tertiary"
+              icon="XMark"
+              size="small"
+              onClick={handleClear}
+              aria-label="Fjern valg"
+            />
+          ) : (
+            <Icon
+              iconName="ChevronDown"
+              className={cl(styles.chevron, { [styles.open]: isOpen })}
+            />
+          )}
```

Comment:
Nothing happens when I click this button. Maybe it needs an onClick event listener to open the select box!